mmdet/datasets/pipelines/transforms.py

```diff
@@ -302,17 +302,29 @@ class RandomFlip(object):
     method.
 
     Args:
-        flip_ratio (float, optional): The flipping probability. Default: None.
-        direction(str, optional): The flipping direction. Options are
-            'horizontal' and 'vertical'. Default: 'horizontal'.
+        flip_ratio (float | list[float], optional): The flipping probability.
+            Default: None.
+        direction(str | list[str], optional): The flipping direction. Options
+            are 'horizontal', 'vertical', 'diagonal'. Default: 'horizontal'.
+            If input is a list, the length must equal ``flip_ratio``. Each
+            element in ``flip_ratio`` indicates the flip probability of
+            corresponding direction.
     """
 
     def __init__(self, flip_ratio=None, direction='horizontal'):
```

Comment:
We need to support 3 cases:
- flip_ratio is a float, direction is a string.
- flip_ratio is a float, direction is a list of strings.
- flip_ratio is a list of float, direction is a list of strings.